src/components/TwoFactorAuthModal.js

```diff
@@ -0,0 +1,176 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ModalWindow from './Modal';
+import QRCode from 'qrcode.react';
+import {injectIntl, FormattedMessage} from 'react-intl';
+import {connect} from 'react-redux';
+import { closeTwoFactorAuthModal, disableTFA, confirmTFA } from '../actions/modal';
+import { ApiTwoFactorAuth } from '../generic/api';
+
+
+const Api2FA = new ApiTwoFactorAuth();
+
+class TwoFactorAuthModal extends React.Component {
+  state = {
+    isInfoModalOpen: this.props.modal.isTwoFactorAuthModalOpen,
+    codeIsWrong: false,
+    currentCode: '',
+    firstAutoSubmit: true,
+  };
+
+  static propTypes = {
+    modal: PropTypes.shape({
+      mode: PropTypes.string,
+    }),
+    onTwoFactorAuthSubmit: PropTypes.func,
+    secret: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.number,
+    ]),
+  };
+
+  static defaultProps = {
+    modal: {},
+    secret: '',
+    onTwoFactorAuthSubmit: () => ({}),
+  };
+
+
+  static getDerivedStateFromProps(nextProps, prevState) {
+    if (nextProps.modal.isTwoFactorAuthModalOpen !== prevState.isInfoModalOpen) {
+      return {
+        isInfoModalOpen: nextProps.modal.isTwoFactorAuthModalOpen
+      };
+    }
+    return null;
+  }
+
+  onChangeAutoSubmit = e => {
+    const { firstAutoSubmit } = this.state;
+    const currentCode = e.target.value;
+    if (currentCode.length <= 6) {
+      this.setState({currentCode, codeIsWrong: false});
+    } else {
+      this.setState({ currentCode: currentCode.slice(0, 6) });
```

Comment:
This will accept invalid input as valid. Use maxlength attribute of the input to limit input value length.